### PR TITLE
Add Antenna.reference_antenna utility function

### PR DESCRIPTION
**Pull request by bmerry**
This is to replace code that is getting copy-pasted all over the place.

### CODE REVIEW

**Comment by mauch** [comment]
We currently have a cal `refant` for gains and a katdal `ref_ant` for activity in older files (these have already caused confusion amongst users).
Could we perhaps rename this to something like `reference_position` to avoid further confusion?

**Comment by bmerry** [comment]
> We currently have a cal refant for gains and a katdal ref_ant for activity in older files (these have already caused confusion amongst users).
> Could we perhaps rename this to something like reference_position to avoid further confusion?

Indeed, "reference" is rather overloaded. I wanted to have "antenna" in the name though to make it clear that you're getting an Antenna back, not just a tuple of coordinates. Maybe `reference_position_antenna`?

**Comment by mauch** [comment]
That should be fine- and I can't think of anything better than that.

File: katpoint/antenna.py [hunk 1]
```diff
@@ -320,3 +320,17 @@ def _scalar_local_sidereal_time(t):
...
+        This is mainly useful as the reference `antenna` for
+        :meth:`.Target.uvw`, in which case it will give both faster and more
+        accurate results than other choices.
+
+        The returned antenna will have no delay or pointing model. It is
+        intended to be used only for its position and does not correspond to a
+        physical antenna.
+        """
+        pos = self.ref_position_wgs84 if self.delay_model else self.position_wgs84
+        return Antenna(name, pos[0], pos[1], pos[2], self.diameter, beamwidth=self.beamwidth)
```
Comment (ludwigschwardt) [inline katpoint/antenna.py]: The beamwidth is probably not needed, but doesn't hurt either. For the record, I've even made reference antennas with diameter=0 in the past. We can leave this as is.

File: katpoint/antenna.py [hunk 2]
```diff
@@ -320,3 +320,17 @@ def _scalar_local_sidereal_time(t):
             return np.array([_scalar_local_sidereal_time(t) for t in timestamp])
         else:
             return _scalar_local_sidereal_time(timestamp)
+
+    def array_reference_antenna(self, name='array'):
+        """Returns a synthetic antenna at the delay model reference position of this antenna.
```
Comment (ludwigschwardt) [inline katpoint/antenna.py]: I would state "Synthetic antenna..." with the "Returns a" bit.

**Pull request merged**